semgrep-core/src/utils/Immutable_buffer.ml

```diff
@@ -0,0 +1,40 @@
+(* Nat Mote
```

Comment:
It would be so nice if we could have all of our commonly used modules in `utils` like this! But that's a problem for another day